packages/catalog-process/src/services/catalogService.ts

```diff
@@ -1033,14 +1057,7 @@ export function updateDescriptorLogic({
       ),
   };
 
-  const filteredDescriptor = eService.data.descriptors.filter(
-    (d: Descriptor) => d.id !== descriptorId
-  );
-
-  const updatedEService: EService = {
-    ...eService.data,
-    descriptors: [...filteredDescriptor, updatedDescriptor],
-  };
```

Comment:
Refactor. In this way the `replaceDescriptor` function (implemented below) is used twice.